packages/catalog-server/src/lib/firestore/firestore-repository.ts

```diff
@@ -441,6 +489,32 @@ export class FirestoreRepository implements Repository {
     return customElementsResults.docs.map((d) => d.data());
   }
 
+  async queryElements({
+    query,
+    limit,
+  }: {
+    query?: string;
+    limit?: number;
+  }): Promise<Array<CustomElement>> {
+    let dbQuery: Query<CustomElement> | CollectionGroup<CustomElement> = db
+      .collectionGroup('customElements')
+      .withConverter(customElementConverter)
+      .where('isLatest', '==', true)
+      .limit(limit ?? 25);
+
+    if (query !== undefined) {
+      // Split query
+      const queryTerms = natural.PorterStemmer.tokenizeAndStem(query);
+      if (queryTerms.length > 10) {
+        queryTerms.length = 10;
+      }
+      dbQuery = dbQuery.where('searchTerms', 'array-contains-any', queryTerms);
```

Comment:
We discussed offline, but just for the record here -- it sounds like the `searchTerms` field is automatically indexed, and firestore supports indexed `array-contains-any` lookups.

Reply:
Yep, and there are no errors in the emulator.